benchmark-service/benchmark/benchmark.py

```diff
@@ -16,6 +20,65 @@
 log = logging.getLogger('benchmark')
 
 
+# storage_client = storage.Client()
+# bucket = storage_client.get_bucket('hail-benchmarks')
+# blob = bucket.get_blob('0.2.20-3b2b439cabf9.json')
+#
+# blob_str = blob.download_as_string(client=None)
+# pre_data = json.loads(blob_str)
+
+filepath = 'tpoterba/0.2.45-ac6815ee857c-master.json' #'/0.2.45-ac6815ee857c-master.json' test
+
+
+def get_benchmarks(file_path):
+    # create storage client
+    storage_client = storage.Client()
+    # get bucket with name
+    bucket = storage_client.get_bucket('hail-benchmarks')
+    try:
+        # get bucket data as blob
+        blob = bucket.blob(file_path)
+        # convert to string
+        json_data = blob.download_as_string()
+        pre_data = json.loads(json_data)
+    except Exception:
+        message = 'could not find file'
+        log.info('could not get blob: ' + message, exc_info=True)
+        raise web.HTTPBadRequest(text=message)
+
+    # with open('file-to-download-to') as file_obj:
+    #     storage_client.download_blob_to_file(file_path, file_obj)
+
+    # with open(file_path) as f:
+    #     pre_data = json.load(f)
+
+    x = re.findall('.*/+(.*)-(.*)-(.*)?\.json', file_path)  #was file_path
```

Comment:
What I'd do in this case if there are multiple formats that can match is either

1. Make two regexes and then check the longer one first to see if that matches and if not check the shorter one. 

2. My preference here would be to use named capture groups and then use the parentheses to group as well. Something like this (you'll need to test this and fix it potentially): `'gs://hail-benchmarks/(?P<user>[^/]+)/)(?P<version>[^-]+)-)(?P<sha>[^-]+)-)(?P<tag>)?\.json'`

This is an example of how we usually do this. Make a compiled REGEX as a global variable.
```
DISK_TYPE_REGEX = re.compile('(projects/(?P<project>[^/]+)/)?zones/(?P<zone>[^/]+)/diskTypes/(?P<disk_type>.+)')
```

and then use it in code similar to this:

```
def parse_disk_type(name):
    match = DISK_TYPE_REGEX.fullmatch(name)
    return match.groupdict()
```